src/context-providers/builder/hook.ts

```diff
@@ -15,12 +15,21 @@ export const useBuilder = () => {
 
     const updateOrderedIdentifiers = useCallback(
         (orderedIdentifiers: IElementIdentifier[]) => {
+            const newElements = {};
+            orderedIdentifiers.map((value) => {
+                newElements[value.internalId] =
+                    state.elements[value.internalId];
+            });
```

Comment:
instead of reordering the elements here, a more complete solution will be to do so in `generateSchema()`

this is because there are other actions updating the ordered identifiers.

though irrelevant now, `.map()` is not appropriate here because we expect a 1-to-1 transformation with it. `.reduce()` would have been more suitable